packages/lib/src/inputGen.ts

```diff
@@ -0,0 +1,105 @@
+const BN = require("bn.js");
+// @ts-ignore
+import { buildPoseidonReference } from "circomlibjs";
+import { EdwardsPoint, WeierstrassPoint, babyjubjub } from "./babyJubjub";
+import { Signature, MerkleProof } from "./types";
+import {
+  hashEdwardsPublicKey,
+  hexToBigInt,
+  publicKeyFromString,
+} from "./utils";
+
+export const generateMerkleProof = async (
+  pubKeys: string[],
+  index: number
+): Promise<MerkleProof> => {
+  const TREE_DEPTH = 10;
+  const ZEROS = [
+    "0",
+    "14744269619966411208579211824598458697587494354926760081771325075741142829156",
+    "7423237065226347324353380772367382631490014989348495481811164164159255474657",
+    "11286972368698509976183087595462810875513684078608517520839298933882497716792",
+    "3607627140608796879659380071776844901612302623152076817094415224584923813162",
+    "19712377064642672829441595136074946683621277828620209496774504837737984048981",
+    "20775607673010627194014556968476266066927294572720319469184847051418138353016",
+    "3396914609616007258851405644437304192397291162432396347162513310381425243293",
+    "21551820661461729022865262380882070649935529853313286572328683688269863701601",
+    "6573136701248752079028194407151022595060682063033565181951145966236778420039",
+  ];
+  const poseidon = await buildPoseidonReference();
```

Comment:
We can pass this in from the frontend

Reply:
Added support to be able to pass in a hash function from the frontend, will benchmark proving time with this